docs/howto/custom-template-tags.txt

```diff
@@ -498,8 +498,78 @@ you see fit:
     {% current_time "%Y-%m-%d %I:%M %p" as the_time %}
     <p>The time is {{ the_time }}.</p>
 
+.. _howto-custom-template-tags-simple-block-tags:
+
+Simple block tags
+-----------------
+
+.. versionadded:: 5.2
+
+.. method:: django.template.Library.simple_block_tag()
+
+If you need to pass a section of rendered template into your custom tag, the
+``simple_block_tag`` helper function of ``django.template.Library`` passes a
+``content`` argument, containing the rendered template content enclosed in the
+tag. This functions similar to :meth:`~django.template.Library.simple_tag()`,
+but with an additional ``content`` argument.
+
+For example, a custom block tag which created a chart could look like this::
+
+    from django import template
+    from myapp.charts import render_chart
+
+    register = template.Library()
+
+
+    @register.simple_block_tag
+    def chart(content):
+        return render_chart(source=content)
+
+The ``content`` argument contains everything in between the ``{% chart %}``
+and ``{% endchart %}`` tags. If there are other template tags or variables,
+they will be rendered before being passed in.
+
+.. code-block:: html+django
+
+    {% chart %}
+        digraph G {
+            label = "Chart for {{ request.user }}"
+            A -> {B C}
+        }
+    {% endchart %}
+
+A few things to note about the ``simple_block_tag`` helper function:
+
+* The first argument must be called ``content``, and will contain the contents
+  of the template tag as a rendered string.
+* Variables passed to the tag are not included in the rendering context of the
+  content, as would be when using the ``{% with %}`` tag.
+
+Just like :ref:`simple_tag<howto-custom-template-tags-simple-tags>`,
+``simple_block_tag``:
+
+* Validates the quantity and quality of the arguments.
+* Strips quotes from arguments if necessary.
+* Escapes the output accordingly.
+* Supports passing ``takes_context=True`` at registration time to access context.
+  Note that in this case, the first argument *must* be called ``context``,
+  and ``content`` must follows.
+* Supports renaming the tag by passing the ``name`` argument when registering.
+* Supports accepting any number of positional or keyword arguments.
+* Supports storing the result in a template variable using the ``as`` variant.
+
+.. admonition::  Content Escaping
+
+    ``simple_block_tag`` behaves similarly to ``simple_tag`` regarding
+    auto-escaping. For details on escaping and safety, refer to ``simple_tag``.
+
+    Because the ``content`` argument has already been rendered by Django, it is
+    already escaped. Therefore, it may be necessary to use
+    :func:`~django.utils.safestring.mark_safe` to ensure it's not
+    double-escaped.
```

Comment:
Does this refer to the following from the `simple_tag` docs?

> Unlike other tag utilities, simple_tag passes its output through conditional_escape() if the template context is in autoescape mode, to ensure correct HTML and protect you from XSS vulnerabilities.
> If additional escaping is not desired, you will need to use mark_safe() if you are absolutely sure that your code does not contain XSS vulnerabilities. For building small HTML snippets, use of format_html() instead of mark_safe() is strongly recommended.

If so, I think it would be worth replacing the sentence above with something similar, because it's much less ambiguous. 

Otherwise, at least I'd suggest using only `format_html` instead of `mark_safe`.

Also, adding some html markup in the inner tag contents to all tests that rely on returning `content` would also help clarify what needs to be escaped and how exactly.

Reply:
No, this is slightly different. The `simple_tag` docs are specifically about the _output_ of the tag, which is still true for `simple_block_tag`.

This point is about the provided `content` string, which is different.

Reply:
So, for example if I add some html markup in the block contents then `end_one_param_block` doesn't render it as html tags because the interpolation is done incorrectly in the tag definition.

But `escape_format_html_block` still works because it uses `format_html`, which calls `conditional_escape` on `content` and returns a safe string. If I do `return content` as is that works too.

These all seems expected to me, so I'm not sure what this note refers to exactly.

(Also, if there's html in the contents then that's NOT "escaped" in `content` but `content` is a safe string that renders as html, so I don't think that's a correct statement either.)

Reply:
I might be still missing something but I think even the whole admonition could be removed because I don't see anything surprising about how this works. The `content` argument is a `SafeString` (as I'd expect it to be, but that might be still worth pointing out) and all the standard django rules about handling those apply here too.

Reply:
@balazs-endresz Thanks. I have pushed a few changes where I add a more complete example that shows how `format_html` is used. Let me know what you think!

Reply:
Thank you, the docs change and the examples look good to me.